### PR TITLE
Add Lengler mutation rate for OnePlusOne

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

**Comment by teytaud** [comment]
@CarolaDoerr

File: nevergrad/optimization/optimizerlib.py [hunk 1]
```diff
@@ -141,7 +141,13 @@ def _internal_ask_candidate(self) -> p.Parameter:
                 if intensity < 1:
                     intensity = 1
                 data = mutator.portfolio_discrete_mutation(pessimistic_data, intensity)
-            elif mutation == "doerr":
+            elif mutation == "lengler":
+                alpha = 1.54468
+                intensity: float = self.dimension * (alpha * np.log(self._num_ask) / self._num_ask)
+                if intensity < 1:
+                    intensity = 1
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:148]:
```suggestion
                intensity = float(max(1.0, self.dimension * (alpha * np.log(self._num_ask) / self._num_ask)))
```

File: nevergrad/optimization/optimizerlib.py [hunk 2]
```diff
@@ -141,7 +141,13 @@ def _internal_ask_candidate(self) -> p.Parameter:
...
+                data = mutator.portfolio_discrete_mutation(pessimistic_data, intensity)
+             elif mutation == "doerr":
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:150]:
```suggestion
            elif mutation == "doerr":
```

File: nevergrad/optimization/optimizerlib.py [hunk 3]
```diff
@@ -210,7 +214,7 @@ class ParametrizedOnePlusOne(base.ConfiguredOptimizer):
         - `"doublefastga"`: double-FastGA mutations from the current best (Doerr et al, Fast Genetic Algorithms, 2017)
         - `"portfolio"`: Random number of mutated bits (called niform mixing in
           Dang & Lehre "Self-adaptation of Mutation Rates in Non-elitist Population", 2016)
-
+        - `"lengler"`
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:217]: @teytaud please update the docstring

**Review by jrapin — CHANGES_REQUESTED** [review]

File: nevergrad/optimization/optimizerlib.py [hunk 4]
```diff
@@ -137,10 +137,14 @@ def _internal_ask_candidate(self) -> p.Parameter:
                 data = mutator.portfolio_discrete_mutation(pessimistic_data, max(1, int(self._adaptive_mr * self.dimension)))
             elif mutation == "discreteBSO":
                 assert self.budget is not None, "DiscreteBSO needs a budget."
-                intensity: int = int(self.dimension - self._num_ask * self.dimension / self.budget)
+                intensity: float = int(self.dimension - self._num_ask * self.dimension / self.budget)
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:140]:
```suggestion
                intensity = int(self.dimension - self._num_ask * self.dimension / self.budget)
```

File: nevergrad/optimization/optimizerlib.py [hunk 5]
```diff
@@ -137,10 +137,14 @@ def _internal_ask_candidate(self) -> p.Parameter:
...
                 if intensity < 1:
                     intensity = 1
                 data = mutator.portfolio_discrete_mutation(pessimistic_data, intensity)
+            elif mutation == "lengler":
+                alpha = 1.54468
+                intensity = float(max(1.0, self.dimension * (alpha * np.log(self._num_ask) / self._num_ask)))
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:146]:
```suggestion
                intensity = float(max(1.0, self.dimension * (alpha * np.log(self.num_ask) / self.num_ask)))
```

File: nevergrad/optimization/optimizerlib.py [hunk 6]
```diff
@@ -137,10 +137,14 @@ def _internal_ask_candidate(self) -> p.Parameter:
...
             elif mutation == "discreteBSO":
                 assert self.budget is not None, "DiscreteBSO needs a budget."
-                intensity: int = int(self.dimension - self._num_ask * self.dimension / self.budget)
+                intensity = int(self.dimension - self._num_ask * self.dimension / self.budget)
                 if intensity < 1:
                     intensity = 1
                 data = mutator.portfolio_discrete_mutation(pessimistic_data, intensity)
+            elif mutation == "lengler":
+                alpha = 1.54468
+                intensity = float(max(1.0, self.dimension * (alpha * np.log(self.num_ask) / self.num_ask)))
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:146]: @teytaud should it be an integer? `portfolio_discrete_mutation` only takes integers (should that be modified?)

Comment (teytaud) [reply nevergrad/optimization/optimizerlib.py]: I'll use int (as you did) for now as maybe this is more meaningful as it is somehow homogeneous to a number of variables, but given the stochastic nature of the method we could discuss this...

Comment (jrapin) [reply nevergrad/optimization/optimizerlib.py]: I used integer because you used integer initially :p it's definitely up to you, as long as it is coherent

**Review by jrapin — APPROVED** [review]

**Pull request merged**